<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>913</width>
    <height>541</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>FudanDNN</string>
  </property>
  <property name="dockOptions">
   <set>QMainWindow::AllowTabbedDocks|QMainWindow::AnimatedDocks</set>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QToolBox" name="unitToolBox">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Maximum" vsizetype="Ignored">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>220</width>
        <height>0</height>
       </size>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="basicUnitBox">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>220</width>
         <height>440</height>
        </rect>
       </property>
       <attribute name="label">
        <string>Basic tools</string>
       </attribute>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="QGraphicsView" name="graphicsView">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QToolBox" name="settingBox">
      <property name="minimumSize">
       <size>
        <width>260</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>250</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="networkMetaInformation">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>260</width>
         <height>414</height>
        </rect>
       </property>
       <property name="cursor">
        <cursorShape>ArrowCursor</cursorShape>
       </property>
       <attribute name="label">
        <string>Network Meta Information</string>
       </attribute>
       <layout class="QFormLayout" name="formLayout">
        <item row="0" column="0">
         <widget class="QLabel" name="networkCriteriaComboLabel">
          <property name="text">
           <string>Criteria:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QComboBox" name="networkCriteriaComboBox">
          <item>
           <property name="text">
            <string>MSE</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Softmax</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="networkBatchSizeSpinBoxLabel">
          <property name="text">
           <string>Batch Size:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSpinBox" name="networkBatchSizeSpinBox">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>1000</number>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="networkTrainingTimesSpinBoxLabel">
          <property name="text">
           <string>Train Time:</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QSpinBox" name="networkTrainingTimesSpinBox">
          <property name="minimum">
           <number>100</number>
          </property>
          <property name="maximum">
           <number>2147483647</number>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="networkOutput">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>122</width>
         <height>205</height>
        </rect>
       </property>
       <attribute name="label">
        <string>Network Output</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QPushButton" name="startTrainingBtn">
          <property name="text">
           <string>Start Training</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="startPredictingBtn">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Start Predicting</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pauseTrainingBtn">
          <property name="text">
           <string>Pause</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="abortTrainingBtn">
          <property name="text">
           <string>Abort</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTextEdit" name="networkOutputText">
          <property name="verticalScrollBarPolicy">
           <enum>Qt::ScrollBarAlwaysOn</enum>
          </property>
          <property name="horizontalScrollBarPolicy">
           <enum>Qt::ScrollBarAlwaysOn</enum>
          </property>
          <property name="undoRedoEnabled">
           <bool>false</bool>
          </property>
          <property name="lineWrapMode">
           <enum>QTextEdit::NoWrap</enum>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>913</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpenNetwork"/>
    <addaction name="actionSaveNetwork"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuExecute">
    <property name="title">
     <string>Execute</string>
    </property>
    <addaction name="actionResetNetwork"/>
    <addaction name="actionResetItemsPosition"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuExecute"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="minimumSize">
    <size>
     <width>200</width>
     <height>0</height>
    </size>
   </property>
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <action name="actionOpenNetwork">
   <property name="text">
    <string>Open Network</string>
   </property>
  </action>
  <action name="actionSaveNetwork">
   <property name="text">
    <string>Save Network</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionTrain">
   <property name="text">
    <string>Train</string>
   </property>
  </action>
  <action name="actionOutputGraph">
   <property name="text">
    <string>Output Graph</string>
   </property>
  </action>
  <action name="actionResetItemsPosition">
   <property name="text">
    <string>Reset Items Position</string>
   </property>
  </action>
  <action name="actionCompileNetwork">
   <property name="text">
    <string>Compile Network</string>
   </property>
  </action>
  <action name="actionResetNetwork">
   <property name="text">
    <string>Clear Network</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="networkeditor.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>414</x>
     <y>270</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
